datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategyFactory.java

```diff
@@ -47,7 +52,15 @@ public AssignmentStrategy createStrategy(Properties assignmentStrategyProperties
         Optional.empty();
     Optional<Integer> partitionFullnessThresholdPct = cfgPartitionFullnessThresholdPct > 0 ?
         Optional.of(cfgPartitionFullnessThresholdPct) : Optional.empty();
+    String cluster = props.getString(CFG_CLUSTER_NAME);
+
+    // Create the ZooKeeper Client
+    String zkAddress = props.getString(CFG_ZK_ADDRESS);
+    int zkSessionTimeout = props.getInt(CFG_ZK_SESSION_TIMEOUT, ZkClient.DEFAULT_SESSION_TIMEOUT);
+    int zkConnectionTimeout = props.getInt(CFG_ZK_CONNECTION_TIMEOUT, ZkClient.DEFAULT_CONNECTION_TIMEOUT);
+    ZkClient zkClient = new ZkClient(zkAddress, zkSessionTimeout, zkConnectionTimeout);
+
     return new StickyPartitionAssignmentStrategy(maxTasks, imbalanceThreshold, maxPartitions,
-        enableElasticTaskAssignment, partitionsPerTask, partitionFullnessThresholdPct);
+        enableElasticTaskAssignment, partitionsPerTask, partitionFullnessThresholdPct, zkClient, cluster);
```

Comment:
zkClient usage has direct ties with enableElasticTaskAssignment. Can we make some checks here itself to enable ElasticTaskAssignment only if properties related to zk client is passed ? The field zkClient should change to `Optional<ZkClient>`.

Reply:
For now I'd like to pass the `enableElasticTaskAssignment` flag as I want it to be explicitly configured rather than just relying on ZK client configs to exist (when copy pasting configs, it's easy to not know the  implication of enabling certain features purely by the existence of other configs that aren't obviously related). Modified the code to pass a valid ZkClient only if `enableElasticTaskAssignment` is true.

Reply:
Got it. Modified the factory to disable it if the zkAddress is not present.